### ISSUE TITLE
Firestore: Using TypeScript `Uppercase<string>` causes incorrect type error

### ISSUE THREAD

**Issue by jinh0** [firebase/firebase-js-sdk#6918]
### [REQUIRED] Describe your environment

  * Operating System version: MacOS 12.1
  * Browser version: Safari 15.2
  * Firebase SDK version: 9.14.0
  * Firebase Product: firestore (auth, database, storage, etc)
  * TypeScript version: 4.8.4

### [REQUIRED] Describe the problem

Using `Uppercase<string>` on a type for a `CollectionReference` causes an error when using updateDoc, when it shouldn't. 

#### Steps to reproduce:

Create a collection reference with an object type with an `Uppercase<string>` attribute. Then try using `updateDoc`, and compile with TypeScript. Get the following error:

```
Type error: Argument of type '{ other: string; }' is not assignable to parameter of type '{ code?: FieldValue | UpdateData<Uppercase<string>>; other?: string | FieldValue; } & UnionToIntersection<ChildUpdateFields<"code", Uppercase<string>>>'.
  Type '{ other: string; }' is not assignable to type 'UnionToIntersection<ChildUpdateFields<"code", Uppercase<string>>>'.
```

#### Relevant Code:

```typescript
type User = {
  code: Uppercase<string>
  other: string
}

let users = collection(firestore, 'users') as CollectionReference<User>

updateDoc(doc(users, 'test-user'), {
  other: 'something',
})```

**Comment by cherylEnkidu** [comment]
Duplicate of https://github.com/googleapis/nodejs-firestore/issues/1745

Hi,

Thanks for filling the ticket! Our team is aware of this issue and is working on the solution.

**Comment by dconeybe** [comment]
Please check if this was fixed in v10.0.0, as there were several fixes to UpdateData typing issues: https://firebase.google.com/support/release-notes/js#version_1000_-_july_6_2023
